R/epichains.R

```diff
@@ -0,0 +1,283 @@
+    writeLines(sprintf("Use View(<object_name>) to view the full output."))
+
+  } else if (attributes(x)$chain_type == "chains_vec") {
+    cat(sprintf("epichains object \n"))
+    print(as.vector(x))
+    cat(sprintf("Number of chains simulated: %s",
+                chain_info[["unique_chains"]]
+                )
+        )
+    writeLines(
+      c(
+        "\n Simulated chain stats: \n",
+        sprintf("Max: %s", chain_info[["max_chain_stat"]]),
+        sprintf("Min: %s", chain_info[["min_chain_stat"]])
+      )
+    )
+  }
+
+  invisible(x)
+}
+
+
+
+#' Summary method for epichains class
+#'
+#' @param object An [`epichains`] object
+#' @param ... further arguments passed to or from other methods
+#'
+#' @return data frame of information
+#' @export
+summary.epichains <- function(object, ...) {
+  validate_epichains(object)
+
+  if (attributes(object)$chain_type == "chains_tree") {
+
+    chains_ran <- length(object$n)
+
+    max_time <- max(object$time)
+
+    n_unique_ancestors <- length(
+      unique(object$ancestor[!is.na(object$ancestor)])
+    )
+
+    num_generations <- length(unique(object$generation))
+
+    max_generation <- max(object$generation)
+
+    # out of summary
+    res <- list(
+      unique_chains = chains_ran,
+      max_time = max_time,
+      unique_ancestors = n_unique_ancestors,
+      unique_generations = n_unique_ancestors,
+      num_generations = num_generations,
+      max_generation = max_generation
+    )
+  } else if (attributes(object)$chain_type == "chains_vec") {
+    chains_ran <- length(object)
+    max_chain_stat <- max(!is.infinite(object))
+    min_chain_stat <- min(!is.infinite(object))
+
+    res <- list(
+      unique_chains = chains_ran,
+      max_chain_stat = max_chain_stat,
+      min_chain_stat = min_chain_stat
+    )
+  }
+
+  return(res)
+}
+
+#' Checks whether the object is an `epichains`
+#'
+#' @param x An R object
+#'
+#' @return logical, `TRUE` if the object is an `epichains` and `FALSE`
+#' otherwise
+#' @keywords internal
+is_epichains <- function(x) {
+  inherits(x, "epichains")
+}
+
+#' Check if an object is of class "epichains_aggregate_df"
+#'
+#' @param x An [`epichains`] object
+#'
+#' @keywords internal
+is_epichains_aggregate_df <- function(x) {
+  if (!inherits(x, "epichains_aggregate_df")) {
+    stop("Object must have class 'epichains_aggregate_df'")
+  }
+}
+
+#' `epichains` class validator
+#'
+#' @param x An `epichains` object
+#'
+#' @return Checks if an object is of class `epichains` and if so
+#' checks that it's in the right format as a "data.frame" or vector.
+#' @keywords internal
+#' @author James M. Azam
+validate_epichains <- function(x) {
+  if (!is_epichains(x)) {
+    stop("Object must have an epichains class")
+  }
+
+  # check for class invariants
+
+  if (attributes(x)$chain_type == "chains_tree") {
+    stopifnot(
+      "object does not contain the correct columns" =
+        c("sim_id", "ancestor", "generation", "time") %in%
```

Comment:
"time" is not strictly necessary - only if used with a serial interval

Reply:
Fixed in 5d3e108769cc6239b2b28abab7c29f9ae02a7163.